lib/teach-api.js

```diff
@@ -38,8 +38,13 @@ util.inherits(TeachAPI, EventEmitter);
 
 _.extend(TeachAPI.prototype, {
   logout: function() {
+    var currentUsername = this.getUsername();
+
     delete this.storage[STORAGE_KEY];
-    this.emit('username:change', null);
+
+    if (currentUsername !== null) {
```

Comment:
Oh, we basically call `logout` from `checkLoginStatus` if the server tells us that nobody is logged in, no matter what, just to ensure that the client-side state (session storage and anyone listening for the `logout` event) is synced with server-side state. (That said, the `logout` and `login:success` *are* potentially being emitted more than they need to be, which for our purposes is OK right now.)